x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/serialize.ts

```diff
@@ -15,35 +15,46 @@ export interface SerializeResult {
   on_failure?: Processor[];
 }
 
-const convertProcessorInternalToProcessor = (processor: ProcessorInternal): Processor => {
-  const { options, onFailure, type } = processor;
+const convertProcessorInternalToProcessor = (
+  processor: ProcessorInternal,
+  addTag?: boolean
+): Processor => {
+  const { options, onFailure, type, id } = processor;
   const outProcessor = {
     [type]: {
       ...options,
     },
   };
 
   if (onFailure?.length) {
-    outProcessor[type].on_failure = convertProcessors(onFailure);
-  } else if (onFailure) {
-    outProcessor[type].on_failure = [];
```

Comment:
@jloleysens can you verify I didn't introduce any regressions by deleting this line? It was causing issues when calling the simulate API, as the `on_failure` field cannot be an empty array.

Reply:
Ah ok, I _think_ this should be fine. The thinking here was that we return to the caller the same set of values that was passed in - so if they passed in an empty array, we would return one. That being said, I am happy to collapse that value because the ES API accepts undefined for that value.

Not accepting an empty array is kinda weird though!